### PR TITLE
Api

### PR DESCRIPTION
**Pull request by SimonLab**
Update the design of the app and deal with the issues:
- [45](https://github.com/FAC-GM/app/issues/45)
- [41](https://github.com/FAC-GM/app/issues/41)
- [39](https://github.com/FAC-GM/app/issues/39)
- [38](https://github.com/FAC-GM/app/issues/38)
- [36](https://github.com/FAC-GM/app/issues/36)
- [33](https://github.com/FAC-GM/app/issues/33)
- [31](https://github.com/FAC-GM/app/issues/31)

### CODE REVIEW

**Comment by SimonLab** [comment]
Sorry it's a big one to review

**Comment by nelsonic** [comment]
Wow...

**Comment by SimonLab** [comment]
The last commit deal with the issue [42](https://github.com/FAC-GM/app/issues/42)

File: .travis.yml [hunk 1]
```diff
@@ -1,8 +1,10 @@
...
-  - 0.10
-  - 0.12
+  - 4.2
 services:
   - elasticsearch
 before_script:
   - sleep 10
+  - curl -XPOST 'localhost:9200/_bulk?pretty' --data-binary "@./test/fixtures/fixture.json"
+after_script:
+    - curl -XDELETE 'localhost:9200/fixture'
```
Comment (nelsonic) [inline .travis.yml]: is there a *reason* you need to delete the fixture data after your tests complete...?

File: assets/css/mobile-first.css [hunk 2]
```diff
@@ -3,8 +3,8 @@
 ================================*/
 html {
   background-color: #f2f2f2;
-  font: 100%/1.3 'Vollkorn', serif;
-  font-family: 'Vollkorn', serif;
+  font: 100%/1.3 'Helvetica', serif;
+  font-family: 'Helvetica', serif;
```
Comment (nelsonic) [inline assets/css/mobile-first.css]: can't go wrong with a bit of [*Helvetica*](http://www.imdb.com/title/tt0847817/) ...

File: lib/api.js [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+
+        es.search({
+          index: process.env.ES_INDEX,
+          body: {
+              query: {
+                  match_phrase: {url: request.payload.url}
+              }
+         }
+
+        }, function (error, response) {
```
Comment (nelsonic) [inline lib/api.js]: in the *next* iteration, you could consider splitting out this function to make it easier to debug.

File: lib/candidateView.js [hunk 4]
```diff
@@ -8,13 +8,13 @@ exports.register = function (server, options, next) {
     config: {
       description: 'return the candidate detailed view page',
       handler: function (request, reply) {
-          
+
           es.get({
-            index: 'globalm',
-            type: 'contacts',
+            index: process.env.ES_INDEX,
```
Comment (nelsonic) [inline lib/candidateView.js]: did you make these Environment Variables because they are different in Local vs. Production?

File: lib/search.js [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+                if (pageNum > 1) {
+                  page_url_prev = '/search/' + encodeURIComponent(request.params.query) + '/' + (pageNum - 1);
+                }
+                if (pageNum < page_url_next) {
+                  pageNum++;
+                  page_url_next = '/search/' + encodeURIComponent(request.params.query) + '/' + pageNum;
+                }
+
+
+                return reply.view('home',
```
Comment (nelsonic) [inline lib/search.js]: this is *very similar* to `lib/home.js` ... we could re-use the home or search view and simply use a different data/result set.

File: test/home.test.js [hunk 6]
```diff
@@ -0,0 +1,126 @@
...
+
+  it('returns the home page', function (done) {
+
+    Server.init(0, function (err, server) {
+
+      expect(err).to.not.exist();
+
+      server.inject('/', function (res) {
+
+        expect(res.statusCode).to.equal(200);
```
Comment (nelsonic) [inline test/home.test.js]: are all these tests `statusCode` checks? what *else* are you expecting for these routes?

File: assets/css/mobile-first.css (hunk 2 above)
Comment (iteles) [inline assets/css/mobile-first.css]: I'm very partial to [Open Sans](https://www.google.com/fonts/specimen/Open+Sans) for body text myself :wink:

File: .travis.yml (hunk 1 above)
Comment (SimonLab) [inline .travis.yml]: Not really, I guess I was thinking to keep a clean elasticsearch after the test. Also I don't use anymore the fixture index

File: lib/candidateView.js (hunk 4 above)
Comment (SimonLab) [inline lib/candidateView.js]: Yes, and if the database need to change we don't have to go to the code to change every values

File: test/home.test.js (hunk 6 above)
Comment (SimonLab) [inline test/home.test.js]: I'm also checking for some redirection (302) on specific cases.

**Comment by nelsonic** [comment]
@heron2014 @SimonLab @iteles I've made a few updates to the Readme to help our *future selves* with running this app. 
The code looks decent. the only thing I would do is improve the tests. lets pair on this in the next sprint. :+1:

File: .travis.yml (hunk 1 above)
Comment (nelsonic) [inline .travis.yml]: Travis runs all tests in a VM and automatically deletes all data when it destroys the VM after tests complete.
(*there is no way of retaining the data from one build to the next...*)

**Comment by iteles** [comment]
@nelsonic Perfect, I was just about to push up my vagrantfile changes! :ok_hand:

**Comment by nelsonic** [comment]
@iteles sorry, I hadn't realised you were editing the Vagrantfile... I streamlined it a bit.

**Pull request merged**